### PR TITLE
[Monorepo] Fix flyteidl generated code and scripts

### PR DESCRIPTION
**Pull request by eapolinario**
## Describe your changes

This PR regenerates the flyteidl files and also ports the flyteidl verification github workflow from the original repo.

## Check all the applicable boxes 

- [x] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -65,8 +63,6 @@ jobs:
           - datacatalog
           - flyteadmin
           - flytecopilot
-          # TODO(monorepo): Enable flyteidl unit tests
```
Comment (eapolinario) [inline .github/workflows/checks.yml:68]: these are now enabled in flyteidl-checks.yml

File: flyteidl/Makefile [hunk 2]
```diff
@@ -7,13 +7,8 @@ define PIP_COMPILE
 pip-compile $(1) --upgrade --verbose
 endef
 
-.PHONY: update_boilerplate
-update_boilerplate:
```
Comment (eapolinario) [inline flyteidl/Makefile:11]: no need to update boilerplate as that comes from monorepo now.

File: flyteidl/boilerplate/flyte/golang_support_tools/go.mod [hunk 3]
```diff
@@ -5,33 +5,17 @@ go 1.19
 require (
 	github.com/EngHabu/mockery v0.0.0-20220405200825-3f76291311cf
 	github.com/alvaroloes/enumer v1.1.2
-	github.com/flyteorg/flytestdlib v0.4.16
```
Comment (eapolinario) [inline flyteidl/boilerplate/flyte/golang_support_tools/go.mod:8]: we build the `pflags` binary beforehand now, so no need to depend on this.

Comment (pingsutw) [reply flyteidl/boilerplate/flyte/golang_support_tools/go.mod:8]: we'll remove boilerplate eventually, right

**Review by pingsutw — APPROVED** [review]

**Review by jeevb — APPROVED** [review]

**Pull request merged**